code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -223,5 +196,128 @@
 			//Prevents whatever mechanism is causing it to hit negative temperatures.
 		return REACTING
 
+/datum/gas_reaction/brownsformation //The formation of brown gas. Endothermic.
+	priority = 3
+	name = "Brown Gas formation"
+	id = "brownsformation"
+
+/datum/gas_reaction/brownsformation/init_reqs()
+	min_requirements = list(
+		"oxygen" = 20,
+		"nitrogen" = 20,
+		"temp" = FIRE_MINIMUM_TEMPERATURE_TO_EXIST*200
+	)
+
+/datum/gas_reaction/brownsformation/react(datum/gas_mixture/air)
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+
+	var/old_heat_capacity = air.heat_capacity()
+	var/heat_efficency = temperature/FIRE_MINIMUM_TEMPERATURE_TO_EXIST*100
+	var/energy_used = heat_efficency*BROWNS_FORMATION_ENERGY
+	air.assert_gases("oxygen","nitrogen","browns")
+
+	cached_gases["oxygen"][MOLES] -= heat_efficency
+	cached_gases["nitrogen"][MOLES] -= heat_efficency
+	cached_gases["browns"][MOLES] += heat_efficency*2
+
+	if(energy_used > 0)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = max(((temperature*old_heat_capacity - energy_used)/new_heat_capacity),TCMB)
+		return REACTING
+
+/datum/gas_reaction/bzformation //Formation of BZ by combining plasma and tritium at low pressures. Exothermic.
+	priority = 3
+	name = "BZ Gas formation"
+	id = "bzformation"
+
+/datum/gas_reaction/bzformation/init_reqs()
+	min_requirements = list(
+		"tritium" = 10,
+		"plasma" = 10
+	)
+
+
+/datum/gas_reaction/bzformation/react(datum/gas_mixture/air)
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/pressure = air.return_pressure()
+
+	var/old_heat_capacity = air.heat_capacity()
+	var/reaction_efficency = pressure/0.1*ONE_ATMOSPHERE
+	var/energy_released = 2*reaction_efficency*FIRE_CARBON_ENERGY_RELEASED
+
+
+	air.assert_gases("tritium","plasma","BZ")
+	cached_gases["bz"][MOLES]+= reaction_efficency
+	cached_gases["tritium"][MOLES]-= 2*reaction_efficency
+	cached_gases["plasma"][MOLES]-= reaction_efficency
```

Comment:
loss of mass intentiona?

Reply:
We don't really know the molar masses of most of the gases in game. But yes.